Stripe/PublicHeaders/STPSourceParams.h

```diff
@@ -0,0 +1,201 @@
+//
+//  STPSourceParams.h
+//  Stripe
+//
+//  Created by Ben Guo on 1/23/17.
+//  Copyright © 2017 Stripe, Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "STPFormEncodable.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class STPCardParams;
+
+/**
+ *  An object representing parameters used to create a Source object. 
+ *  @see https://stripe.com/docs/api#create_source
+ */
+@interface STPSourceParams : NSObject<STPFormEncodable>
+
+/**
+ *  The type of the source to create. Required.
+ */
+@property (nonatomic, copy, nullable) NSString *type;
+
+/**
+ *  A positive integer in the smallest currency unit representing the
+ *  amount to charge the customer (e.g., @1099 for a €10.99 payment).
+ *  Required for `single_use` sources.
+ */
+@property (nonatomic, copy, nullable) NSNumber *amount;
+
+/**
+ *  The currency associated with the source. This is the currency for which the source 
+ *  will be chargeable once ready.
+ */
+@property (nonatomic, copy, nullable) NSString *currency;
+
+/**
+ *  The authentication flow of the source to create. `flow` may be "redirect",
+ *  "receiver", "verification", or "none". It is generally inferred unless a type
+ *  supports multiple flows.
+ */
+@property (nonatomic, copy, nullable) NSString *flow;
+
+/**
+ *  A set of key/value pairs that you can attach to a source object.
+ */
+@property (nonatomic, copy, nullable) NSDictionary *metadata;
```

Comment:
I'd prefer keeping them nullable – if we made them nonnull/mutable dictionaries, I think our serialization logic (`STPFormEncoder`) would add empty dictionaries, rather than leaving out the key pairs.